### ISSUE TITLE
Blob Batch middleware in subrequest incorrectly interpreting account name

### ISSUE THREAD

**Issue by robherley** [Azure/Azurite#2195]
### Which service(blob, file, queue, table) does this issue concern?

blob

### Which version of the Azurite was used?

3.26.0

### Where do you get Azurite? (npm, DockerHub, NuGet, Visual Studio Code Extension)

DockerHub `sha256:f3e3a7eb821d7ca50a2c8f64cee2e2f759096dd69191e5065a385da46d1252c6`

### What's the Node.js version?

v14.21.3 (from `node -v` in the container)

### What problem was encountered?

I'm using Azurite for some integration tests, and I noticed some odd behavior with the [Blob Batch API](https://learn.microsoft.com/en-us/rest/api/storageservices/blob-batch?tabs=azure-ad) when using a hostname instead of an IP address.

Related to:
- https://github.com/Azure/Azurite/issues/1039 (specifically [this comment](https://github.com/Azure/Azurite/issues/1039#issuecomment-916588540))

The subrequests will fail when calling delete with a hostname with `ResourceNotFound` because it's incorrectly calculating the account and container names.

For example, for the request:

```
2023-09-27T02:30:27.862Z 561549c8-b4c5-4ef5-8a59-00c52115bc09 info: BlobStorageContextMiddleware: RequestMethod=DELETE RequestURL=http://devstoreaccount1.blob.localhost:52183/repro/foo/bar/file2.txt RequestHeaders:{"authorization":"SharedKey devstoreaccount1:xoZAFkNZysW1hfk8PLb2PyhX4RcYAvUaJbcki3pGr6M=","x-ms-date":"Wed, 27 Sep 2023 02:30:27 GMT","accept":"application/xml"} ClientIP=http://devstoreaccount1.blob.localhost Protocol=http HTTPVersion=version
```

The middleware is incorrectly interpreting the URL and thinks the first piece of the URL path is the storage account:

```
2023-09-27T02:30:27.862Z 561549c8-b4c5-4ef5-8a59-00c52115bc09 info: BlobStorageContextMiddleware: Account=repro Container=foo Blob=bar/file2.txt
2023-09-27T02:30:27.862Z 561549c8-b4c5-4ef5-8a59-00c52115bc09 verbose: DispatchMiddleware: Dispatching request...
2023-09-27T02:30:27.862Z 561549c8-b4c5-4ef5-8a59-00c52115bc09 info: DispatchMiddleware: Operation=Blob_Delete
2023-09-27T02:30:27.862Z 561549c8-b4c5-4ef5-8a59-00c52115bc09 verbose: AuthenticationMiddlewareFactory:createAuthenticationMiddleware() Validating authentications.
2023-09-27T02:30:27.862Z 561549c8-b4c5-4ef5-8a59-00c52115bc09 info: PublicAccessAuthenticator:validate() Start validation against public access.
2023-09-27T02:30:27.862Z 561549c8-b4c5-4ef5-8a59-00c52115bc09 debug: PublicAccessAuthenticator:validate() Getting account properties...
2023-09-27T02:30:27.862Z 561549c8-b4c5-4ef5-8a59-00c52115bc09 debug: PublicAccessAuthenticator:validate() Retrieved account name from context: repro, container: foo, blob: bar/file2.txt
2023-09-27T02:30:27.863Z 561549c8-b4c5-4ef5-8a59-00c52115bc09 debug: PublicAccessAuthenticator:validate() Skip public access authentication. Cannot get public access type for container foo
2023-09-27T02:30:27.863Z 561549c8-b4c5-4ef5-8a59-00c52115bc09 info: BlobSharedKeyAuthenticator:validate() Start validation against account shared key authentication.
2023-09-27T02:30:27.863Z 561549c8-b4c5-4ef5-8a59-00c52115bc09 error: BlobSharedKeyAuthenticator:validate() Invalid storage account repro.
```
([full debug log here](https://gist.github.com/robherley/143b649a6382f686c0aca9d2d1a72196#file-zzz_debug-log))

However this works perfectly fine for all other APIs as far as I can tell. It's only in these `DELETE` subrequests for the batch api that I've observed this.

### Steps to reproduce the issue?

I have a snippet using the Azure SDK for Go that also spins up a fresh azurite container:
- https://gist.github.com/robherley/143b649a6382f686c0aca9d2d1a72196

You'll need to have the docker daemon running and just run `go run main.go`.

I have a few parameters at the top of the file to help enumerate all the scenarios:

```go
// Toggle between IP and Hostname
UseIP = false
// Toggle between batch API and one-by-one delete
UseBatchDelete = true
```

The above case is the failure condition. If `UseIP` is `true`, the process will complete fine, since the account name is interpreted correctly. If the `UseBatchDelete` is false, it will also complete fine by calling delete blob one by one instead of the batch API. The issue only occurs with the blob batch APIs.

The code also creates a `log/` directory in cwd and will write the Azurite debug log to it.

### Have you found a mitigation/solution?

Nope

**Comment by blueww** [comment]
@EmmaZhu 

Would you please help to look at the blob batch issue?

**Comment by EmmaZhu** [comment]
Hi @robherley , 

I have sent a PR here to try to resolve this issue: https://github.com/Azure/Azurite/pull/2214 . I have tested the change with JS SDK.  Not sure whether you are familiar with building Azurite. If you are, could you help to validate in your environment to see whether this can resolve your issue?

**Comment by robherley** [comment]
@EmmaZhu I can confirm your fix works 🎉 Thank you!

```
### Using URL: http://devstoreaccount1.blob.localhost:10000
2023/10/09 22:45:02 uploading "foo/bar/file1.txt" with content "This is file 1"
2023/10/09 22:45:02 uploading "foo/bar/file2.txt" with content "This is file 2"
2023/10/09 22:45:02 uploading "foo/bar/file3.txt" with content "This is file 3"
2023/10/09 22:45:02 - foo/bar/file1.txt
2023/10/09 22:45:02 - foo/bar/file2.txt
2023/10/09 22:45:02 - foo/bar/file3.txt
2023/10/09 22:45:02 adding "foo/bar/file1.txt" to delete batch
2023/10/09 22:45:02 adding "foo/bar/file2.txt" to delete batch
2023/10/09 22:45:02 - foo/bar/file3.txt
```

**Comment by EmmaZhu** [comment]
The fix has been included in the latest release azurite@3.27.0.  If the issue still exists, please feel free to reopen this issue.
